planner/core/planbuilder.go

```diff
@@ -309,7 +309,11 @@ func (b *PlanBuilder) popSelectOffset() {
 
 // NewPlanBuilder creates a new PlanBuilder.
 func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor *BlockHintProcessor) *PlanBuilder {
-	sctx.GetSessionVars().PlannerSelectBlockAsName = make([]model.CIStr, processor.MaxSelectStmtOffset()+1)
+	if processor == nil {
```

Comment:
Is this a bug fix?

Reply:
https://github.com/pingcap/tidb/pull/12358  add a new feature to resolve column name, and this change need planBuilder to pass in valid `hint processor`. The earlier point execution optimization pass in nil `hint processor` and skip related logic,  and the unit test case does not cover this path,  so after https://github.com/pingcap/tidb/pull/12358 merged, sysbench will panic.  #12482 is already fixing this, this pull request make unit test cover these paths